File: java/src/main/java/deltix/luminary/UInt32Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class UInt32Def implements TypeDef<TypeUInt32>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private UInt32Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeUInt32.INSTANCE, new LiteralUInt32(0x00_00_00_00)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeUInt32.INSTANCE, new LiteralUInt32(0xFF_FF_FF_FF)));
    }

    public static final UInt32Def INSTANCE = new UInt32Def();

    @Override
    public String getName() {
        return "UInt32";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeUInt32 getType() {
        return TypeUInt32.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}

File: csharp/main/UInt32Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class UInt32Def : IConstantScope, ITypeDef<TypeUInt32>
	{
		private UInt32Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeUInt32.Instance, new LiteralUInt32(UInt32.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeUInt32.Instance, new LiteralUInt32(UInt32.MaxValue)));
		}

		public static UInt32Def Instance = new UInt32Def();

		public String Name => "UInt32";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeUInt32 Type => TypeUInt32.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}
